framework/include/interfaces/BlockRestrictable.h

```diff
@@ -202,6 +209,11 @@ class BlockRestrictable
    */
   virtual void checkVariable(const MooseVariableFieldBase & variable) const;
 
+  /**
+   * Adds blocks to the block restriction of the object
+   */
+  void addBlocks(const std::vector<SubdomainName> & new_blocks);
```

Comment:
this is potentially controversial so it's in its own commit. 
happy to separate it if needed

Reply:
This is a pretty simple design. An alternative could be to make a "MutableBlockRestrictable" interface which inherits from BlockRestrictable and is used only for objects we know benefit from a modified BR (e.g. mostly variables! but maybe the NS Rhie Chow user objects too)

Reply:
ok it also does not work as the variable block restriction is given to the system at creation time. 
Maybe that can be changed. But for now I m removing this commit in the interest of getting this merged

Reply:
yea definitely would prefer to remove this until it's determined it's needed

Reply:
ok done!